launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/TTSServlet.java

```diff
@@ -0,0 +1,245 @@
+        } catch (Exception e) {
+            res.sendError(HttpServletResponse.SC_CONFLICT, "Unable to translate. Try again");
+            return;
+        }
+
+        // Synthesize The podcast Text
+        ByteString synthesizedMp3 = null;
+        try {
+            synthesizedMp3 = synthesizeText(podcastText);
+        } catch (Exception e) {
+            res.sendError(HttpServletResponse.SC_CONFLICT, "unable to create mp3 from request. Please try again.");
+            return;
+        }
+        Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
+        String itemCount = String.valueOf(rssFeed.getChannel().getItems().size());
+        BlobId blobId = BlobId.of(BUCKET_NAME, feedKey + itemCount);
+        BlobInfo blobInfo = BlobInfo.newBuilder(blobId).build();
+
+        // create an uploadble array of bytes for blobstore
+        byte[] mp3Bytes = synthesizedMp3.toByteArray();
+        storage.create(blobInfo, mp3Bytes);
+
+        // Generate mp3 link
+        String mp3Link = TTS_BASE_URL + feedKey + itemCount; // creates unique ID for each episode
+
+        rssFeed.getChannel().addItem(podcastTitle, podcastDescription, podcastLanguage, userEmail, mp3Link);
+
+        desiredFeedEntity.setProperty(XML_STRING, RSS.toXmlString(rssFeed)); // update existing entity's XML string
+
+        datastore.put(desiredFeedEntity);
+
+        Query query = new Query(LoginStatus.USER_FEED_KEY).addSort(LoginStatus.TIMESTAMP_KEY, SortDirection.DESCENDING);
+
+        PreparedQuery results = datastore.prepare(query);
+
+        ArrayList<UserFeed> userFeeds = new ArrayList<UserFeed>();
+        queryUserFeeds(userFeeds, results, userEmail);
+
+        res.setContentType("application/json");
+        res.getWriter().println(GSON.toJson(userFeeds));
+    }
+
+    /**
+     * doGet obtains an ID from the link to search for specific blob in cloud
+     * storage. After obtaining the blob, we get the bytes then stream that to the
+     * client using the server outputstream
+     * 
+     * @throws IOException when unable to run doGet method
+     */
+    public void doGet(HttpServletRequest req, HttpServletResponse res) throws IOException {
+        res.setContentType("audio/mpeg");
+        String id = req.getParameter(ID);
+
+        if (Strings.isNullOrEmpty(id)) {
+            throw new IllegalArgumentException("Invalid URL. Please try again");
+        }
+
+        // Search for id in cloud storage from parameter ID
+        Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
+        Blob desiredFeedBlob = storage.get(BUCKET_NAME, id);
+        byte[] blobBytes = desiredFeedBlob.getContent(BlobSourceOption.generationMatch());
+
+        // Write bytes to servlet output stream
+        res.getOutputStream().write(blobBytes);
+    }
+
+    public void queryUserFeeds(ArrayList<UserFeed> userFeeds, PreparedQuery results, String userEmail) {
```

Comment:
let's return ArrayList<UserFeed> instead of passing it in as a parameter

Reply:
sounds good!